R/generate_retrospective_data.R

```diff
@@ -0,0 +1,99 @@
+#' Generate retrospective reporting triangles
+#'
+#' This function ingests a reporting triangle and the number of retrospective
+#'   reporting triangles we want to create, and iteratively generates the
+#'   reporting triangle that would have been available as of the maximum
+#'   reference time, and iteratively working backwards for
+#'   `n_history_uncertainty` snapshots
+#'
+#' @param triangle_for_retro Matrix of the incomplete reporting triangle
+#'   to be used to generate retrospective nowcasts, with rows representing the
+#'   time points of reference and columns representing the delays
+#' @param n_history_uncertainty Integer indicating the number of retrospective
+#'   reporting triangles to be generated, always starting from the most
+#'   recent reference time. The default is to generate as many retrospective
+#'   reporting triangles that would have a sufficient number of rows to
+#'   estimate  a delay distribution and generate a nowcast from.
+#' @param n_history_delay Integer indicating the number of reference dates to be
```

Comment:
is this the right name for this? As this is just producing data it might make sense to just call it n_reference or similar?

Reply:
related to above with the filtering and checking to ensure you have enough data to do all of these things. Also trying to use the same parameter names for the same things, so in the `get_delay_estimte()`, we use the same variable to specify the number of observations to use to estimate the delay distribution (must be greater than or equal to the max delay)

Reply:
I feel like the checks should be in the steps where those things happen?

Reply:
I think the difference in thought here is I am thinking about chaining methods whereas having this do the checks is a composition argument I think. My push back would be these tasks aren't really nested and so they don't need to be composed together in a joint function

Reply:
This is updated and no longer contains these checks/specifications